avro-converter/src/main/java/io/confluent/connect/avro/AvroData.java

```diff
@@ -1058,12 +1058,44 @@ private Object toConnectData(Schema schema, Object value) {
             arrayVal =
             recordValue.get(ANYTHING_SCHEMA.getField(ANYTHING_SCHEMA_ARRAY_FIELD).pos());
         if (arrayVal != null) {
-          return toConnectData(SchemaBuilder.array(null).build(), arrayVal);
+          // We cannot reuse the logic like we do in other cases because it is not possible to
+          // construct an array schema with a null item schema, but we do not know the schema of the
+          // items.
+          if (!(arrayVal instanceof Collection)) {
```

Comment:
This is existing code it's just a minor modification of the version where you have schemas. So I'd be inclined to keep both of them aligned.

If anything, the `List` is probably incorrect and technically a bug in the handling of maps since https://avro.apache.org/docs/1.8.1/api/java/org/apache/avro/generic/package-summary.html specifies that arrays are of type `Collection` (true for both `GenericRecord` and `SpecificRecord`. In practice it's probably always a `List` which is why the map code works fine. More generally, where possible I think the more general type is better when accepting input.